### PR TITLE
flake8-pyi: fix PYI015 false positive on assignment of TypeVar & friends

### LINKED ISSUE #3860: flake8-pyi: PYI015 false positive on TypeVar assignment
Given this pyi file:

```py
from typing import TypeVar
_T = TypeVar('_T')
```

running

```
ruff --select PYI015 --isolated --no-cache x.pyi
```

gives

```
x.pyi:2:6: PYI015 [*] Only simple default values allowed for assignments
```

I think `TypeVar` assignments should be excluded from this rule as they appear naturally in pyi files.

Possible workaround is `_T: TypeVar = TypeVar('_T')`, however that's unnecessarily redundant, and pyright doesn't like it.

Similar cases - `TypeAlias` (already handled), `NewType`, `TypeVarTuple`, `ParamSpec`

From a quick test, this doesn't happen on flake8-pyi.

I think I can provide a PR for this.

ruff 0.0.260

### PR DESCRIPTION
**Pull request by bluetech**
Fix #3860.

### CODE REVIEW

File: crates/ruff/src/rules/flake8_pyi/rules/simple_defaults.rs [hunk 1]
```diff
@@ -221,6 +221,23 @@ fn is_valid_default_value_with_annotation(
     false
 }
 
+pub fn is_type_var_call(checker: &Checker, expr: &Expr) -> bool {
```
Comment (bluetech) [inline crates/ruff/src/rules/flake8_pyi/rules/simple_defaults.rs:224]: This is based on a similar helper in pep8-naming https://github.com/charliermarsh/ruff/blob/924bebbb4a0e1a9002a0274cbb58fc85335089dc/crates/ruff/src/rules/pep8_naming/helpers.rs#L57

I considered factoring it out but decided against it for now, I'm not sure they're exactly the same semantically.

**Comment by twoertwein** [comment]
I like the typeshed diff :) Should `ParamSpec` also be whitelisted (not sure how flake8-pyi handles that)?

**Review by charliermarsh — COMMENTED** [review]
I think we might be deviating from `flake8-pyi` in the case of non-annotated assignments. They seem to allow _most_ values when assigning to a variable without a type annotation:

```py
def _is_valid_default_value_without_annotation(node: ast.expr) -> bool:
    """Is `node` a valid default for an assignment without an annotation?"""
    return (
        isinstance(
            node, (ast.Call, ast.Name, ast.Attribute, ast.Subscript, ast.Ellipsis)
        )
        or _is_None(node)
        or _is_valid_pep_604_union(node)
    )
```

I think we should have the same behavior -- what do you think?

**Comment by bluetech** [comment]
> I think we should have the same behavior -- what do you think?

In practice this lint forced me to add `TypeAlias` to a bunch of unannotated assignments, and the rest were TypeVar-like things which are covered by this PR. So it was good. Not sure if there are other reasons to assign without annotation. I guess we can check this lint on typeshed to find out (I assume typeshed is flake8-pyi clean). Will do it when I get a chance.

cc @JonathanPlasse who implemented PYI015 in ruff.

**Comment by AlexWaygood** [comment]
> I assume typeshed is flake8-pyi clean

Indeed it is!

---

@charliermarsh is correct that we handle this pretty differently in flake8-pyi. We use Y026 to enforce that anything that looks like a type alias assignment is explicitly annotated using `TypeAlias`, and Y052 to enforce type annotations for all assignments to literals. With those invariants out of the way, that means that we basically don't have to worry about whether the remaining assignments without type annotations have "valid default values" or not (because there's a surprising number of calls etc. that are valid in stubs, and it doesn't really feel like it would provide much value to work out a comprehensive list), and so Y011/Y014/Y015 only really worry about the validity of default values for assignments that have type annotations.

**Pull request merged**